xbmc/games/agents/GameAgentManager.cpp

```diff
@@ -143,6 +173,24 @@ float CGameAgentManager::GetPortActivation(const std::string& portAddress) const
   return activation;
 }
 
+float CGameAgentManager::GetPeripheralActivation(const std::string& peripheralLocation) const
+{
+  float activation = 0.0f;
+
+  std::lock_guard<std::mutex> lock(m_agentMutex);
+
+  for (const GameAgentPtr& agent : m_agents)
+  {
+    if (agent->GetPeripheralLocation() == peripheralLocation)
+    {
+      activation = agent->GetActivation();
```

Comment:
I like to avoid early returns so that there's a single path out of the function. If we returned early here, we would also have a return at the end of the function.

That said, I like the for-if-break pattern less, so I'll add the early return.